### PR TITLE
Update to current proto-lens packages.

### PR DESCRIPTION
**Pull request by mikesperber**

### CODE REVIEW

File: tensorflow-core-ops/tensorflow-core-ops.cabal [hunk 1]
```diff
@@ -16,7 +16,7 @@ library
   exposed-modules:  TensorFlow.GenOps.Core
   autogen-modules:  TensorFlow.GenOps.Core
   build-depends:  bytestring
-                , proto-lens >= 0.4.0 && < 0.6.0
```
Comment (blackgnezdo) [inline tensorflow-core-ops/tensorflow-core-ops.cabal:19]: Could you retain the safety net for future incompatible changes by changing to keep < 0.7.0 everywhere?

**Comment by blackgnezdo** [comment]
The build failed:
```
ensorflow                 > 
tensorflow                 > /tfhs/tensorflow/src/TensorFlow/Types.hs:79:53: error:
tensorflow                 >     Module ‘Lens.Family2’ does not export ‘under’
tensorflow                 >    |
tensorflow                 > 79 | import Lens.Family2 (Lens', view, (&), (.~), (^..), under)
tensorflow                 >    |                                                     ^^^^^
tensorflow                 > 
tensorflow                 > /tfhs/tensorflow/src/TensorFlow/Types.hs:80:32: error:
tensorflow                 >     Module ‘Lens.Family2.Unchecked’ does not export ‘adapter’
tensorflow                 >    |
tensorflow                 > 80 | import Lens.Family2.Unchecked (adapter)
tensorflow                 >    |                                ^^^^^^^
tensorflow-opgen           > copy/register
tensorflow-opgen           > Installing library in /tfhs/.stack-work/install/x86_64-linux/197511ac4b1d5fecc6a9a155d9f0f87795bb2c470d4024f0e3210eadeb16328d/8.6.5/lib/x86_64-linux-ghc-8.6.5/tensorflow-opgen-0.2.0.1-591Nwo9lUZC1wH2GzvH8bV
tensorflow-opgen           > Registering library for tensorflow-opgen-0.2.0.1..

--  While building package tensorflow-0.2.0.1 using:
      /root/.stack/setup-exe-cache/x86_64-linux/Cabal-simple_mPHDZzAJ_2.4.0.1_ghc-8.6.5 --builddir=.stack-work/dist/x86_64-linux/Cabal-2.4.0.1 build lib:tensorflow test:FFITest test:VarIntTest --ghc-options ""
    Process exited with code: ExitFailure 1

```

File: tensorflow-core-ops/tensorflow-core-ops.cabal (hunk 1 above)
Comment (mikesperber) [reply tensorflow-core-ops/tensorflow-core-ops.cabal:19]: Good point: Done. (By fixing to 0.6.*.)

**Comment by mikesperber** [comment]
@blackgnezdo Ah, that's because there was no version bound for `lens-family`. (Which is strange, as that's the difference between the older and the newer proto-lens mainly.) Fixed.

**Comment by blackgnezdo** [comment]
Looks like the build plan isn't viable :(
```
Error: While constructing the build plan, the following exceptions were encountered:

In the dependencies for tensorflow-0.2.0.1:
    lens-family-1.2.3 from stack configuration does not match ==2.*  (latest matching version
                      is 2.1.0)
    proto-lens-0.5.1.0 from stack configuration does not match ==0.6.*  (latest matching version
                       is 0.6.0.0)
needed since tensorflow is a build target.

In the dependencies for tensorflow-core-ops-0.2.0.1:
    lens-family-1.2.3 from stack configuration does not match ==2.*  (latest matching version
                      is 2.1.0)
    proto-lens-0.5.1.0 from stack configuration does not match ==0.6.*  (latest matching version
                       is 0.6.0.0)
needed since tensorflow-core-ops is a build target.

In the dependencies for tensorflow-logging-0.2.0.1:
    lens-family-1.2.3 from stack configuration does not match ==2.*  (latest matching version
                      is 2.1.0)
    proto-lens-0.5.1.0 from stack configuration does not match ==0.6.*  (latest matching version
                       is 0.6.0.0)
needed since tensorflow-logging is a build target.

In the dependencies for tensorflow-mnist-0.1.0.0:
    lens-family-1.2.3 from stack configuration does not match >=2.0  (latest matching version
                      is 2.1.0)
    proto-lens-0.5.1.0 from stack configuration does not match ==0.6.*  (latest matching version
                       is 0.6.0.0)
needed since tensorflow-mnist is a build target.

In the dependencies for tensorflow-opgen-0.2.0.1:
    lens-family-1.2.3 from stack configuration does not match ==2.*  (latest matching version
                      is 2.1.0)
    proto-lens-0.5.1.0 from stack configuration does not match ==0.6.*  (latest matching version
                       is 0.6.0.0)
needed since tensorflow-opgen is a build target.

In the dependencies for tensorflow-ops-0.2.0.1:
    lens-family-1.2.3 from stack configuration does not match ==2.*  (latest matching version
                      is 2.1.0)
    proto-lens-0.5.1.0 from stack configuration does not match ==0.6.*  (latest matching version
                       is 0.6.0.0)
needed since tensorflow-ops is a build target.

In the dependencies for tensorflow-proto-0.2.0.1:
    proto-lens-0.5.1.0 from stack configuration does not match ==0.6.*  (latest matching version
                       is 0.6.0.0)
    proto-lens-protobuf-types-0.5.0.0 from stack configuration does not match ==0.6.*  (latest
                                      matching version is 0.6.0.0)
    proto-lens-runtime-0.5.0.0 from stack configuration does not match ==0.6.*  (latest matching
                               version is 0.6.0.0)
    proto-lens-setup-0.4.0.2 from stack configuration does not match ==0.6.* 
needed since tensorflow-proto is a build target.

Some different approaches to resolving this:

  * Set 'allow-newer: true'
    in /root/.stack/config.yaml to ignore all version constraints and build anyway.

  * Recommended action: try adding the following to your extra-deps in /tfhs/stack.yaml:

- lens-family-2.1.0@sha256:de40e0281e3410a4a742b748dd4054f3ff6694c2398638da6b95f593b400aa1e,1989
- proto-lens-0.6.0.0@sha256:381073e7442260d3a0f6bf8fdb7371af1967927c2b20b535fcc64433bc8d055d,3036
- proto-lens-protobuf-types-0.6.0.0@sha256:a60c9d64744904ed51488789d9715247bbc87d1af1d1d124bbc2c74b62e4b7a7,2828
- proto-lens-runtime-0.6.0.0@sha256:3aa983a9b968b708e632889e044c34418cbf2fc15b4300f31d50f5cae1dca477,3139

Plan construction failed.
The command '/bin/sh -c apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 575159689BEFB442 &&     apt-get update &&     apt-get install -y         libsnappy-dev         libncurses5-dev         libgmp-dev         libcurl3-dev         zlib1g-dev         curl         unzip         netbase         &&     curl -O -L https://github.com/commercialhaskell/stack/releases/download/v2.1.3/stack-2.1.3-linux-x86_64.tar.gz &&     tar zxf stack-2.1.3-linux-x86_64.tar.gz -C /usr/local/bin stack-2.1.3-linux-x86_64/stack --strip 1 &&     curl -O -L https://github.com/google/protobuf/releases/download/v3.9.1/protoc-3.9.1-linux-x86_64.zip &&     unzip -d /usr/local protoc-3.9.1-linux-x86_64.zip bin/protoc &&     chmod 755 /usr/local/bin/protoc &&     curl -O https://storage.googleapis.com/tensorflow/libtensorflow/libtensorflow-cpu-linux-x86_64-1.14.0.tar.gz &&     tar zxf libtensorflow-cpu-linux-x86_64-1.14.0.tar.gz -C /usr/local &&     ldconfig &&     stack setup &&     stack test --only-dependencies' returned a non-zero code: 1


[ID: 8634961] Build finished after 239 secs, exit value: 1


Warning: Permanently added 'localhost' (ECDSA) to the list of known hosts.

[10:54:38] Collecting build artifacts from build VM
Build script failed with exit code: 1
[10:55:25] Running blaze commands.
[10:55:28] Finished running blaze commands.
[10:55:29] /tmp/workspace/workspace/tensorflow_haskell/github/presubmit/artifacts/custom_sponge_config.csv does not exist in the workspace, no custom config value to log
[10:55:34] Updating SpongeV2 invocation.
[10:55:39] Build does not log to ResultStore. Skipping ResultStore update. Note that Bazel may still log to ResultStore.
```

**Comment by mikesperber** [comment]
@blackgnezdo I'm sorry, I <facepalm> forgot you build with stack, I just used cabal. This requires a bump in LTS. I've updated the patch -is this acceptable?

**Comment by blackgnezdo** [comment]
My apologies, the CI turn-around is terrible. These are the errors:

```
tensorflow-proto           > copy/register
tensorflow-proto           > Installing library in
/tfhs/.stack-work/install/x86_64-linux/89b5e1fbecff86add17122dda36d0acb9493f7c0b5c653d4ba1efe511ff9168f/8.8.3/lib/x86_64-linux-ghc-8.8.3/tensorflow-proto-0.2.0.1-5C60xZETDG81O6OD4hMlHe
tensorflow-proto           > Registering library for
tensorflow-proto-0.2.0.1..
tensorflow-opgen           > configure (lib)
tensorflow-opgen           > Configuring tensorflow-opgen-0.2.0.1...
tensorflow                 > configure (lib + test)
tensorflow-opgen           > build (lib)
tensorflow-opgen           > Preprocessing library for
tensorflow-opgen-0.2.0.1..
tensorflow-opgen           > Building library for tensorflow-opgen-0.2.0.1..
tensorflow-opgen           > [1 of 2] Compiling TensorFlow.OpGen.ParsedOp
tensorflow-opgen           >
tensorflow-opgen           >
/tfhs/tensorflow-opgen/src/TensorFlow/OpGen/ParsedOp.hs:28:1: error:
[-Wunused-imports, -Werror=unused-imports]
tensorflow-opgen           >     The import of ‘Data.Monoid’ is redundant
tensorflow-opgen           >       except perhaps to import instances from
‘Data.Monoid’
tensorflow-opgen           >     To import instances alone, use: import
Data.Monoid()
tensorflow-opgen           >    |
tensorflow-opgen           > 28 | import Data.Monoid ((<>))
tensorflow-opgen           >    | ^^^^^^^^^^^^^^^^^^^^^^^^^
tensorflow                 > Configuring tensorflow-0.2.0.1...
tensorflow                 > build (lib + test)
tensorflow                 > Preprocessing library for tensorflow-0.2.0.1..
tensorflow                 > Building library for tensorflow-0.2.0.1..
tensorflow                 > [ 1 of 12] Compiling TensorFlow.Internal.Raw
tensorflow                 > [ 2 of 12] Compiling TensorFlow.Internal.FFI
tensorflow                 > [ 3 of 12] Compiling TensorFlow.Internal.VarInt
tensorflow                 >
tensorflow                 >
/tfhs/tensorflow/src/TensorFlow/Internal/VarInt.hs:31:1: error:
[-Wunused-imports, -Werror=unused-imports]
tensorflow                 >     The import of ‘Data.Monoid’ is redundant
tensorflow                 >       except perhaps to import instances from
‘Data.Monoid’
tensorflow                 >     To import instances alone, use: import
Data.Monoid()
tensorflow                 >    |
tensorflow                 > 31 | import Data.Monoid ((<>))
tensorflow                 >    | ^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Comment by mikesperber** [comment]
@blackgnezdo Fixed. Could you give me a hint what exactly is run by CI, so I can avoid hassling you with incomplete pull requests?

**Comment by blackgnezdo** [comment]
> @blackgnezdo Fixed. Could you give me a hint what exactly is run by CI, so I can avoid hassling you with incomplete pull requests?

I should've done this before, sorry. The script that it runs is here: https://github.com/tensorflow/haskell/blob/master/ci_build/outer_launch_tests.sh

**Comment by mikesperber** [comment]
@blackgnezdo Thanks for the hint. I've updated the patch and verified that it passes CI.

**Comment by blackgnezdo** [comment]
Thanks for persevering besides our convoluted CI setup!

**Pull request merged**